deb/openmediavault/usr/share/openmediavault/confdb/migrations.d/conf_6.5.0.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env dash
+#
+# This file is part of OpenMediaVault.
+#
+# @license   http://www.gnu.org/licenses/gpl.html GPL Version 3
+# @author    Volker Theile <volker.theile@openmediavault.org>
+# @copyright Copyright (c) 2009-2023 Volker Theile
+#
+# OpenMediaVault is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# any later version.
+#
+# OpenMediaVault is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with OpenMediaVault. If not, see <http://www.gnu.org/licenses/>.
+
+set -e
+
+. /usr/share/openmediavault/scripts/helper-functions
+
+# Convert the enabled NFS versions into a comma separated list.
+# E.g: +2 -3 +4 -4.1 -4.2 => 2,4
+versions=$(cat /proc/fs/nfsd/versions | sed -E 's/-([[:digit:]](.[[:digit:]])?)//g' | tr -d '+' | sed -e 's/[[:space:]]*$//' | tr -s ' ' ',')
```

Comment:
Any improvement to simplify this conversion is welcome.

Reply:
sed -e 's/+//g' -e 's/\s/,/g' /proc/fs/nfsd/versions

Reply:
Thx, but the disabled (-X.Y) protocols need to be removed, too.

Reply:
Ah, I missed that since I don't have any disabled on my system.

Reply:
This is the simplest I could come up

awk '{ for (i=1; i<=NF; i++) if ($i > 0) printf "%s,", $i }' /proc/fs/nfsd/versions | sed -e 's/+//g' -e 's/,$//'

Reply:
Uff, that also looks not that simple :-)